upgrades/upgrade_testing_framework/steps/step_start_source_cluster.py

```diff
@@ -0,0 +1,29 @@
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+from upgrade_testing_framework.cluster_management.cluster import Cluster, ClusterNotStartedInTimeException
+
+class StartSourceCluster(FrameworkStep):
+    """
+    This step starts the source cluster and ensures it is healthy
+    """
+
+    def _run(self):
+        # Get the state we need
+        source_docker_image = self._get_state_value("test_config")["source_docker_image"]
+        docker_client = self.state.docker_client
+
+        # Begin the step body
+        self.logger.info("Creating source cluster...")
+        cluster = Cluster("source-cluster", 3, docker_client, source_docker_image)
+        cluster.start()
+
+        try: 
+            max_wait_sec = 30
+            self.logger.info(f"Waiting up to {max_wait_sec} sec for cluster to be active...")
+            cluster.wait_for_cluster_to_start_up(wait_limit_sec=30)
+        except ClusterNotStartedInTimeException as exception:
```

Comment:
What are your thoughts on cleaning up created resources on failed start ups? Maybe this is a flag we can add in the configuration that some users may want and some may not

Reply:
Nice-to-have, but not required-right-now.  I talk about this a bit in my response to Mikayla a few lines down.  TBH, there's a lot of nice-to-haves that I'm deliberately postponing so that we can iterate quickly to hit our Demo goals.